Gems/Blast/Code/Source/Components/BlastFamilyComponent.cpp

```diff
@@ -191,41 +188,42 @@ namespace Blast
     {
         AZ_PROFILE_FUNCTION(Physics);
 
-        AZ_Assert(m_blastAsset.GetId().IsValid(), "BlastFamilyComponent created with invalid blast asset.");
+        if (!m_blastAsset.GetId().IsValid())
+        {
+            AZ_Warning("BlastFamilyComponent", false, "Blast Family Component created with invalid blast asset.");
+            return;
+        }
 
         Spawn();
-
-        BlastFamilyDamageRequestBus::MultiHandler::BusConnect(GetEntityId());
-        BlastFamilyComponentRequestBus::Handler::BusConnect(GetEntityId());
     }
 
     void BlastFamilyComponent::Deactivate()
     {
         AZ_PROFILE_FUNCTION(Physics);
 
-        // cleanup collision handlers
-        for (auto& itr : m_collisionHandlers)
-        {
-            itr.second.Disconnect();
-        }
-        m_collisionHandlers.clear();
-
-        BlastFamilyDamageRequestBus::MultiHandler::BusDisconnect();
-        BlastFamilyComponentRequestBus::Handler::BusDisconnect();
-
         Despawn();
     }
 
     void BlastFamilyComponent::Spawn()
     {
         AZ_PROFILE_FUNCTION(Physics);
 
-        if (!m_blastAsset.IsReady())
+        if (m_isSpawned)
         {
-            m_shouldSpawnOnAssetLoad = true;
             return;
         }
 
+        if (!m_blastAsset.IsReady())
+        {
+            m_blastAsset.QueueLoad();
+            m_blastAsset.BlockUntilLoadComplete();
+            if (!m_blastAsset.IsReady())
+            {
+                AZ_Error("BlastFamilyComponent", false, "Failed to load blast asset %s.", m_blastAsset.GetHint().c_str());
+                return;
+            }
+        }
+
```

Comment:
This fix is temporary. Now that the Spawn/Despawn code has been improved and it's self-contained (nothing else is done in Activate/Deactivate) it'll be easier to wait for the AssetReady bus notification is received. But it will be done is a follow-up PR.